src/intervals/construction.jl

```diff
@@ -259,31 +253,275 @@ function atomic(::Type{T}, a::AbstractFloat) where {T<:AbstractFloat}
     if a == hi
         hi = nextfloat(hi)
     end
-    return unsafe_interval(T, lo, hi)
+    return _unsafe_bareinterval(T, lo, hi)
+end
+
+
+
+# decorations
+
+"""
+    Decoration
+
+Enumeration constant for the types of interval decorations described in
+Section 11.2 of the IEEE Standard 1788-2015:
+- `com -> 4`: non-empty, continuous and bounded (common)
+- `dac -> 3`: non-empty and continuous (defined and continuous)
+- `def -> 2`: non-empty (defined)
+- `trv -> 1`: always true (trivial)
+- `ill -> 0`: not an interval (ill-formed)
+"""
+@enum Decoration ill=0 trv=1 def=2 dac=3 com=4
+# note: `isless`, and hence `<`, `min` and `max`, are automatically defined
+
+function decoration(x::BareInterval)
+    isnai(x) && return ill
+    isempty_interval(x) && return trv
+    isunbounded(x) && return dac
+    return com
 end
 
-# prevents multiple threads from calling setprecision() concurrently; it is used
-# in `bigequiv`
-const precision_lock = ReentrantLock()
+
+
+
+
+# decorated intervals
 
 """
-    bigequiv(x::Interval)
-    bigequiv(x::Union{Rational,AbstractFloat})
+    Interval{T<:NumTypes} <: Real
+
+Interval type for guaranteed computation with interval arithmetic according to
+the IEEE Standard 1788-2015. This structure combines a [`BareInterval`](@ref)
+together with a [`Decoration`](@ref). The decoration records the validity of the
+operations that produced this interval.
+
+Fields:
+- `bareinterval :: BareInterval{T}`
+- `decoration   :: Decoration`
 
-Create a `BigFloat` equivalent with the same underlying precision as `x`.
+Constructors compliant with the IEEE Standard 1788-2015:
+- [`interval`](@ref)
+- [`..`](@ref)
+- [`±`](@ref)
+- [`@I_str`](@ref)
+
+See also: [`±`](@ref), [`..`](@ref) and [`@I_str`](@ref).
 """
-function bigequiv(x::Interval{T}) where {T<:NumTypes}
-    lock(precision_lock) do
-        setprecision(precision(float(T))) do
-            return Interval{BigFloat}(x)
-        end
-    end
+struct Interval{T<:NumTypes} <: Real
+    bareinterval :: BareInterval{T}
+    decoration   :: Decoration
+    guarantee    :: Bool
```

Comment:
[bikeshedding] I would call that `guaranteed`.

Reply:
[More bikeshedding] How about `isguaranteed`?